frontend/src/metabase/nav/components/SearchBar.tsx

```diff
@@ -151,10 +162,10 @@ function SearchBar({
           onKeyPress={handleInputKeyPress}
           ref={searchInput}
         />
-        {isSmallScreen() && hasSearchText && (
-          <ClearIconButton onClick={onClear}>
+        {isSmallScreen() && isActive && (
+          <CloseSearchButton onClick={handleClickOnClose}>
             <Icon name="close" />
-          </ClearIconButton>
+          </CloseSearchButton>
```

Comment:
We stop using `onClear` callback in this PR, please remove the callback itself in `SearchBar`